src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -221,6 +235,30 @@ public AgentSettings LoadSettings()
                 }
             }
 
+            // See if the server supports our OAuth key exchange for credentials
+            if (agent.Authorization != null && 
+                agent.Authorization.ClientId != Guid.Empty &&
+                agent.Authorization.AuthorizationUrl != null)
+            {
+                var credentialData = new CredentialData
+                {
+                    Scheme = Constants.Configuration.OAuth,
+                    Data =
+                    {
+                        { "clientId", agent.Authorization.ClientId.ToString("D") },
+                        { "authorizationUrl", agent.Authorization.AuthorizationUrl.AbsoluteUri },
+                    },
+                };
+
+                // Save the negotiated OAuth credential data
+                _store.SaveCredential(credentialData);
```

Comment:
The full support isn't in until 100, including updating an existing agent with the 'replace' configuration option. New agent configuration is supported in 99, but the protocol had to be changed after reading through the security requirements.